tests/integration/security/authorization_test.go

```diff
@@ -119,78 +115,80 @@ func TestAuthorization_JWT(t *testing.T) {
-					Request: connection.Checker{
-						From: a,
-						Options: echo.CallOptions{
-							Target:   target,
-							PortName: "http",
-							Scheme:   scheme.HTTP,
-							Path:     path,
-						},
-					},
-					Jwt:           jwt,
-					ExpectAllowed: expectAllowed,
-				}
-			}
-			cases := []rbacUtil.TestCase{
-				newTestCase(b, "[NoJWT]", "", "/token1", false),
-				newTestCase(b, "[NoJWT]", "", "/token2", false),
-				newTestCase(b, "[Token1]", jwt.TokenIssuer1, "/token1", true),
-				newTestCase(b, "[Token1]", jwt.TokenIssuer1, "/token2", false),
-				newTestCase(b, "[Token2]", jwt.TokenIssuer2, "/token1", false),
-				newTestCase(b, "[Token2]", jwt.TokenIssuer2, "/token2", true),
-				newTestCase(b, "[Token1]", jwt.TokenIssuer1, "/tokenAny", true),
-				newTestCase(b, "[Token2]", jwt.TokenIssuer2, "/tokenAny", true),
-				newTestCase(b, "[PermissionToken1]", jwt.TokenIssuer1, "/permission", false),
-				newTestCase(b, "[PermissionToken2]", jwt.TokenIssuer2, "/permission", false),
-				newTestCase(b, "[PermissionTokenWithSpaceDelimitedScope]", jwt.TokenIssuer2WithSpaceDelimitedScope, "/permission", true),
-				newTestCase(b, "[NoJWT]", "", "/tokenAny", false),
-				newTestCase(c, "[NoJWT]", "", "/somePath", true),
-
-				// Test condition "request.auth.principal" on path "/valid-jwt".
-				newTestCase(d, "[NoJWT]", "", "/valid-jwt", false),
-				newTestCase(d, "[Token1]", jwt.TokenIssuer1, "/valid-jwt", true),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/valid-jwt", true),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1WithAud, "/valid-jwt", true),
-
-				// Test condition "request.auth.presenter" on suffix "/presenter".
-				newTestCase(d, "[Token1]", jwt.TokenIssuer1, "/request/presenter", false),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1, "/request/presenter", false),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/request/presenter-x", false),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/request/presenter", true),
-
-				// Test condition "request.auth.audiences" on suffix "/audiences".
-				newTestCase(d, "[Token1]", jwt.TokenIssuer1, "/request/audiences", false),
-				newTestCase(d, "[Token1WithAzp]", jwt.TokenIssuer1WithAzp, "/request/audiences", false),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1WithAud, "/request/audiences-x", false),
-				newTestCase(d, "[Token1WithAud]", jwt.TokenIssuer1WithAud, "/request/audiences", true),
+			for _, srcCluster := range ctx.Clusters() {
+				ctx.NewSubTest(fmt.Sprintf("From a In %s", srcCluster.Name())).Run(func(ctx framework.TestContext) {
+					a := apps.A.Match(echo.InCluster(srcCluster).And(echo.NameSpace(ns.Name())))
+					callCount := 1
+					b := apps.B.Match(echo.NameSpace(ns.Name()))
+					c := apps.C.Match(echo.NameSpace(ns.Name()))
+					d := apps.D.Match(echo.NameSpace(ns.Name()))
+					if ctx.Clusters().IsMulticluster() {
+						// so we can validate all clusters are hit
```

Comment:
how is this validated? Is it done out of the test case here?

Reply:
Validation is done within https://github.com/istio/istio/pull/28088/files#diff-8f18b4c496e45a8c03665ecd21ebfb1a5b62aeeea7c64ef4711265bd5d23d2a1R82, part of every test case if the test runs with multicluster context.